include/zephyr/kernel/internal/sched_priq.h

```diff
@@ -51,8 +63,8 @@ struct k_thread *z_priq_rb_best(struct _priq_rb *pq);
  * to represent their requirements.
  */
 struct _priq_mq {
-	sys_dlist_t queues[32];
-	unsigned int bitmask; /* bit 1<<i set if queues[i] is non-empty */
+	sys_dlist_t queues[PRI_QUEUE_BITMAP_SIZE * BIT_NUM];
+	unsigned int bitmask[PRI_QUEUE_BITMAP_SIZE]; /* bit 1<<i set if queues[i] is non-empty */
```

Comment:
Also, while you're in here: it would be good to:

* Make this a stdint type, which is the current Zephyr style, and not an "unsigned int"
* Allow for 64 bit platforms to use uint64_t here (and __builtin_clzll() of course) for efficiency.

Reply:
Is there a method that I can know my machine is 32bit or 64 bit?

Reply:
`uintptr_t` would do it for both.

Reply:
or `long`

Reply:
Done.